ruby/ql/lib/codeql/ruby/ast/Module.qll

```diff
@@ -23,6 +24,20 @@ class Module extends TModule {
   /** Gets an `include`d module. */
   Module getAnIncludedModule() { result = getAnIncludedModule(this) }
 
+  /** Gets the super class or an included or prepended module. */
```

Comment:
This change is very similar to https://github.com/github/codeql/pull/10747/files#diff-9333de60c5c1b8283b484ae3b7e3cbef24b7bf52aee695df420d0e3f4d55403b 

The main difference is that you use `*` instead of `+` which would make all modules ancestors of themselves, which is odd (even though it is handy in some cases).

@hvitved suggested a `pragma[inline]` annotation on `getAnAncestor` , perhaps you should do that as well.

Reply:
> which would make all modules ancestors of themselves, which is odd (even though it is handy in some cases)

I think it's quite common to define things like this, for example, a type is usually considered a subtype of itself. Also a module is technically on its own ancestor chain.
```rb
module A
end
A.ancestors # [A]
```

I also tend to prefer `*` because you can easily get the `+` variant by doing `getADirectAncestor().getAnAncestor()`. 

However, if the predicate only exposes the `+` variant, you have to do something like `[x, x.getAnAncestor()]` to get the `*` variant, and even that doesn't work if `x` has type `Class`. Either way it can't appear in the middle of an expression chain, which is important iMO.

Reply:
You're right. I somehow thought that `A.ancestors` would not be reflexive.